server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -494,4 +536,16 @@ private static int getTotalShardCount(ClusterState state, Index index) {
         return indexMetaData.getNumberOfShards() * (1 + indexMetaData.getNumberOfReplicas());
     }
 
+    /**
+     * @return Generates a {@link ClusterBlock} that blocks read and write operations on soon-to-be-closed indices. The
+     * cluster block is generated with the id value equals to {@link #INDEX_CLOSED_BLOCK_ID} and a unique UUID.
+     */
+    public static ClusterBlock createIndexClosedBlock() {
+        final String description = "Index is blocked due to on-going index closing operation. Note that the closing process can take " +
```

Comment:
this suggests that there is always an active operation trying to close the index, which we know isn't in case of failures. I wonder if we should say something along the lines of "index preparing to close. Reopen the index to allow reads and writes again or retry closing the index to fully close the index".

Reply:
Ok